File: DocManagementBackend/Models/document.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DocManagementBackend.Models {
    public class Document
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int CreatedByUserId { get; set; }
        [ForeignKey("CreatedByUserId")]
        public required User CreatedBy { get; set; }
        public int TypeId { get; set; }
        [ForeignKey("TypeId")]
        public DocumentType? DocumentType { get; set; }
        public int? CircuitId { get; set; }
        [ForeignKey("CircuitId")]
        public Circuit? Circuit { get; set; }
        public int? CurrentCircuitDetailId { get; set; }
        [ForeignKey("CurrentCircuitDetailId")]
        public CircuitDetail? CurrentCircuitDetail { get; set; }
        public bool IsCircuitCompleted { get; set; } = false;
        [Required]
        public string DocumentKey { get; set; } = string.Empty;
        public string DocumentAlias { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string? Content { get; set; }
        [Required]
        public int Status { get; set; } // 0 = Open, 1 = Validated
        public DateTime DocDate { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public int LigneCouter { get; set; } = 0;
        public bool IsDeleted { get; set; } = false;
        [JsonIgnore]
        public ICollection<Ligne> Lignes { get; set; } = new List<Ligne>();
    }

    public class DocumentType
    {
        [Key]
        public int Id { get; set; }
        public string TypeKey { get; set; } = string.Empty;
        public string TypeName { get; set; } = string.Empty;
        public string TypeAttr { get; set; } = string.Empty;
        public int DocumentCounter { get; set; } = 0;
        [JsonIgnore]
        public ICollection<Document> Documents { get; set; } = new List<Document>();
    }

    public class TypeCounter
    {
        public int Id { get; set; }
        public int Counter { get; set; }
        public int circuitCounter { get; set; }
    }
    public class DocumentCircuitHistory
    {
        [Key]
        public int Id { get; set; }

        public int DocumentId { get; set; }
        [ForeignKey("DocumentId")]
        // [DeleteBehavior(DeleteBehavior.NoAction)] // Change this from Cascade
        public required Document Document { get; set; }

        public int CircuitDetailId { get; set; }
        [ForeignKey("CircuitDetailId")]
        // [DeleteBehavior(DeleteBehavior.NoAction)] // Change this from Cascade
        public required CircuitDetail CircuitDetail { get; set; }

        public int ProcessedByUserId { get; set; }
        [ForeignKey("ProcessedByUserId")]
        // [DeleteBehavior(DeleteBehavior.NoAction)] // Change this from Cascade
        public required User ProcessedBy { get; set; }

        public string Comments { get; set; } = string.Empty;
        public DateTime ProcessedAt { get; set; } = DateTime.UtcNow;
        public bool IsApproved { get; set; }
    }
}

File: DocManagementFrontend/src/models/document.ts
import { ResponsibilityCentreSimple } from './responsibilityCentre';
import { 
  LignesElementType, 
  Item, 
  UniteCode, 
  GeneralAccounts 
} from './lineElements';
import { SubType } from './subtype';
import './circuit.d.ts';

export interface Document {
  id: number;
  title: string;
  documentKey: string;
  content: string;
  status: number;
  documentAlias: string;
  documentExterne?: string;
  createdAt: string;
  updatedAt: string;
  typeId: number;
  subTypeId?: number;
  docDate: string;
  comptableDate: string;
  documentType: DocumentType;
  subType?: SubType;
  circuitId?: number;
  circuit?: Circuit;
  currentCircuitDetailId?: number;
  currentCircuitDetail?: CircuitDetail;
  createdByUserId: number;
  createdBy: DocumentUser;
  updatedByUserId?: number;
  updatedBy?: DocumentUser;
  lignesCount?: number;
  sousLignesCount?: number;
  lignes?: Ligne[];
  responsibilityCentreId?: number;
  responsibilityCentre?: ResponsibilityCentreSimple;
}

export interface DocumentType {
  id?: number;
  typeName: string;
  typeKey?: string;
  typeAttr?: string;
  documentCounter?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface DocumentUser {
  id: number;
  username: string;
  firstName: string;
  lastName: string;
  role: string;
  email: string;
}

export interface CreateDocumentRequest {
  title: string;
  content: string;
  documentAlias?: string;
  documentExterne?: string;
  typeId: number;
  subTypeId?: number | null;
  docDate?: string;
  circuitId?: number;
  responsibilityCentreId?: number;
}

export interface UpdateDocumentRequest {
  title?: string;
  content?: string;
  documentAlias?: string;
  documentExterne?: string;
  typeId?: number;
  docDate?: string;
  comptableDate?: string;
  circuitId?: number;
}

export interface Ligne {
  id: number;
  documentId: number;
  ligneKey: string;
  title: string;
  article: string;
  
  // Legacy field
  prix: number;
  
  sousLignesCount: number;
  
  // Type information - FIXED field names to match backend
  lignesElementTypeId?: number;
  lignesElementType?: LignesElementType;
  
  // Element references
  itemCode?: string;
  item?: Item;
  generalAccountsCode?: string;
  generalAccounts?: GeneralAccounts;
  
  // Pricing fields
  quantity: number;
  priceHT: number;
  discountPercentage: number;
  discountAmount?: number;
  vatPercentage: number;
  
  // Calculated fields
  amountHT: number;
  amountVAT: number;
  amountTTC: number;
  
  createdAt: string;
  updatedAt: string;
  document?: Document;
  sousLignes?: SousLigne[];
}

export interface CreateLigneRequest {
  documentId: number;
  ligneKey?: string;
  title: string;
  article: string;
  
  // Type and element references - FIXED field names
  lignesElementTypeId?: number;
  itemCode?: string;
  generalAccountsCode?: string;
  
  // Pricing fields
  quantity: number;
  priceHT: number;
  discountPercentage: number;
  discountAmount?: number;
  vatPercentage: number;
}

export interface UpdateLigneRequest {
  ligneKey?: string;
  title?: string;
  article?: string;
  
  // Type and element references - FIXED field names
  lignesElementTypeId?: number;
  itemCode?: string;
  generalAccountsCode?: string;
  
  // Pricing fields
  quantity?: number;
  priceHT?: number;
  discountPercentage?: number;
  discountAmount?: number;
  vatPercentage?: number;
}

export interface SousLigne {
  id: number;
  ligneId: number;
  sousLigneKey?: string;
  title: string;
  attribute: string;
  createdAt?: string;
  updatedAt?: string;
  ligne?: Ligne;
}

export interface CreateSousLigneRequest {
  ligneId: number;
  title: string;
  attribute: string;
}

export interface UpdateSousLigneRequest {
  title?: string;
  attribute?: string;
}
